scipy/stats/_continuous_distns.py

```diff
@@ -1611,6 +1611,20 @@ def _sf(self, x, a):
                         0.5*sc.gammaincc(a, x),
                         0.5 + 0.5*sc.gammainc(a, -x))
 
+    def _entropy(self, a):
+        #shape = np.shape(a)
+        #a = np.atleast_1d(a)
+        if a < 1e15:
+            h = a + np.log(2) + sc.gammaln(a) + (1 - a) * sc.digamma(a)
+
+        else:
+            h = np.log(2) + 0.5 * (1 + np.log(a) + np.log(2 * np.pi))
```

Comment:
All of scipy's distributions support numpy arrays in their methods. We have to make sure that if a user provides a NumPy array of several values, the correct branch is chosen. Throughout this file, there is a convenience function called `_lazywhere` we should use here too.

The cutoff should also be lowered. For `1e13`, the regular formula becomes very imprecise already.

Reply:
Done.